iam.yml

```diff
@@ -10,13 +10,159 @@ Parameters:
   TemplateBucket:
     Type: String
     Description: Name of existing S3 bucket for storing CloudFormation template artifacts
+  JavabuilderApiId:
+    Type: String
+    Description: The API ID of the Javabuilder API Gateway V2 Api
+    Default: "*"
 
 Resources:
-  # TODO: Split this into separate roles:
-  # 1) Permissions for the long-running Lambda that compiles/runs student projects
-  # 2) Permissions for the synchronous Lambda that invokes the long-running Lambda and then relays web messages to it.
-  # 3) Permissions for the Lambda Authorizer
-  # 4) Permissions for CodePipeline Slack approval Lambda.
```

Comment:
nit: could we something like this around as a comment describing the new separate roles along with their role names?

Reply:
I've added comments describing each role to add more context